src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php

```diff
@@ -260,12 +264,13 @@ public function prepare_item_for_response( $theme, $request ) {
 				$data[ $field ]['rendered'] = $theme->display( $header );
 			}
 		}
+
 		$current_theme = wp_get_theme();
 		if ( rest_is_field_included( 'status', $fields ) ) {
-			$data['status'] = ( $this->is_current_theme( $theme, $current_theme ) ) ? 'active' : 'inactive';
+			$data['status'] = ( $this->is_same_theme( $theme, $current_theme ) ) ? 'active' : 'inactive';
 		}
 
-		if ( rest_is_field_included( 'theme_supports', $fields ) && $this->is_current_theme( $theme, $current_theme ) ) {
+		if ( rest_is_field_included( 'theme_supports', $fields ) ) {
```

Comment:
Theme supports can only be added on the current theme as theme supports requires the theme to be active.

Reply:
Thanks @spacedmonkey - as we discussed I am gonna modify this to ensure that we always return `theme_supports` for consistencies purposes, but for inactive themes it will be an empty array.